### ISSUE TITLE
pytest == 5.1.2 raises ModuleNotFoundError on Windows

### ISSUE THREAD

**Issue by laura-surcel** [pytest-dev/pytest#5857]
Hello!

When using latest pytest version (pytest 5.1.2) on Windows, I encountered an issue when pytest is trying to load a `conftest.py` module found in a folder with both uppercase and lowercase letters.

Stacktrace:
```
ImportError while loading conftest 'd:\pkg\tests\mng\conftest.py'.
ModuleNotFoundError: No module named 'tests.mng'
```

The file on disk is at the path: `D:\pkg\tests\Mng\conftest.py`

The error appears when running any pytest command in the D:\pkg directory. Example:

`D:\pkg>pytest --collect-only tests`

I tested with both pytest 5.1.1 and with 5.1.2.
pytest 5.1.1: no issues
pytest 5.1.2: ModuleNotFoundError

pip list:
Package                       Version                         Location
----------------------------- ------------------------------- ---------------------------
amqp                          1.4.9
anyjson                       0.3.3
APScheduler                   3.5.3
asn1crypto                    0.24.0
astroid                       2.2.5
atomicwrites                  1.3.0
attrs                         19.1.0
backcall                      0.1.0
backpack                      0.22.0
backports.functools-lru-cache 1.5
beautifulsoup4                4.8.0
billiard                      3.3.0.23
bleach                        3.1.0
celery                        3.1.26.post2
celery-with-redis             3.0
certifi                       2019.9.11
cffi                          1.12.3
chardet                       3.0.4
check-manifest                0.39
cheroot                       6.5.8
CherryPy                      17.4.2
Click                         7.0
colander                      1.7.0
colorama                      0.4.1
contextlib2                   0.5.5
cornice                       1.0.0
coverage                      4.5.4
cryptography                  2.7
decorator                     4.4.0
devpi-client                  2.3.0
devpi-common                  3.4.0
elasticsearch-provisioner     0.10.0
enum34                        1.1.6
filelock                      3.0.12
flume-logger                  1.13.0
future                        0.17.1
gitdb2                        2.0.5
GitPython                     2.1.13
graphviz                      0.13
hiredis                       1.0.0
hupper                        1.8.1
idna                          2.8
importlib-metadata            0.19
ipaddress                     1.0.22
ipdb                          0.12.2
ipython                       7.7.0
ipython-genutils              0.2.0
iso8601                       0.1.12
isort                         4.3.21
jaraco.functools              2.0
jedi                          0.14.1
Jinja2                        2.10.1
jsonschema                    2.6.0
kazoo                         2.6.1
lazy-object-proxy             1.4.1
lxml                          4.4.1
MarkupSafe                    1.1.1
mccabe                        0.6.1
MiniMock                      1.2.8
mock                          3.0.5
more-itertools                5.0.0
msgpack                       0.6.1
mysqlclient                   1.4.2
nose                          1.3.7
nose-exclude                  0.5.0
objgraph                      3.4.1
packaging                     19.1
parameterized                 0.7.0
parso                         0.5.1
PasteDeploy                   2.0.1
pbr                           5.4.3
pem                           19.2.0
pickleshare                   0.7.5
pip                           19.2.3
pip-tools                     4.0.0
pkginfo                       1.5.0.1
plaster                       1.0
plaster-pastedeploy           0.7
pluggy                        0.12.0
plumbum                       1.6.7
portend                       2.5
prompt-toolkit                2.0.9
psutil                        5.6.3
py                            1.8.0
pycparser                     2.19
pycryptodomex                 3.9.0
pycurl                        7.43.0.2
PyDispatcher                  2.0.5
Pygments                      2.4.2
pylint                        2.3.1
pymongo                       3.9.0
pyOpenSSL                     19.0.0
pyparsing                     2.4.2
pyramid                       1.10.4
pyreadline                    2.1
pytest                        5.1.2
pytest-mock                   1.10.4
python-dateutil               2.8.0
pytz                          2019.2
pywin32                       225
PyYAML                        3.11
pyzmq                         18.1.0
redis                         2.10.6
release-me                    0.18.1
pkg                    3.32.1.dev10+g2437e44.d20190917 d:\pkg
repoze.lru                    0.6
requests                      2.22.0
setuptools                    41.2.0
setuptools-scm                1.17.0
simplejson                    3.8.2
six                           1.12.0
smmap2                        2.0.5
soupsieve                     1.9.3
SQLAlchemy                    1.3.8
sqlalchemy-migrate            0.12.0
sqlparse                      0.3.0
Tempita                       0.5.2
tempora                       1.14.1
thrift                        0.11.0
toml                          0.10.0
tox                           3.13.2
traitlets                     4.3.2
translationstring             1.3
typed-ast                     1.4.0
tzlocal                       2.0.0
urllib3                       1.25.3
validators                    0.10.1
venusian                      1.2.0
virtualenv                    16.7.2
waitress                      1.3.1
wcwidth                       0.1.7
web.py                        0.40.dev1
webencodings                  0.5.1
WebOb                         1.8.5
WebTest                       2.0.33
wheel                         0.33.6
wincertstore                  0.2
wrapt                         1.11.2
xmltodict                     0.11.0
zc.lockfile                   2.0
zipp                          0.5.2
zope.deprecation              4.4.0
zope.interface                4.6.0

Windows version: Windows 10 Enterprise 1809

**Comment by blueyed** [comment]
https://github.com/pytest-dev/pytest/issues/5819 ?

**Comment by Zac-HD** [comment]
Yep, this has been fixed by #5840 and 5.1.3 should be unaffected.  Sorry for the inconvenience @laura-surcel!
